extensions/amp-list/0.1/amp-list.js

```diff
@@ -48,10 +54,35 @@ export class AmpList extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    return this.populateList_();
+  }
+
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const srcMutation = mutations['src'];
+    if (srcMutation != undefined) {
+      const p = this.populateList_();
+      if (getMode().test) {
+        this.mutationPromise_ = p;
```

Comment:
100% agree that this code isn't ideal. I try to avoid depending on private properties/methods when writing unit tests, hence the promise and the `visibleForTesting` getter. If you think that changing it as proposed will make this easier to read / work with let me know and I'll make the change. 👍